homeassistant/components/pi_hole/config_flow.py

```diff
@@ -150,20 +135,72 @@ async def async_step_reauth_confirm(
             errors=errors,
         )
 
-    async def _async_try_connect(self) -> dict[str, str]:
+    def _api_by_version(
```

Comment:
Seems like this could be factored out of `setup_entry` and reused here?

Reply:
Done :)